tests/tools/test_logs.py

```diff
@@ -1,641 +1,450 @@
+                {"since": "today"},
+                ["--since", "today"],
+                ["since=today"],
+            ),
+            # Custom line count
+            (
+                {"lines": 50},
+                ["-n", "50"],
+                ["last 50 entries"],
+            ),
+            # All filters combined
+            (
+                {"unit": "nginx.service", "priority": "err", "since": "today", "lines": 50},
+                ["-u", "nginx.service", "-p", "err", "--since", "today", "-n", "50"],
+                ["last 50 entries", "unit=nginx.service", "priority=err", "since=today"],
+            ),
+        ],
+    )
+    async def test_get_journal_logs_filters(self, mock_execute_command, params, expected_args, expected_in_output):
+        """Test get_journal_logs with various filter combinations."""
         mock_execute_command.return_value = (
             0,
-            "Jan 01 13:00:00 host nginx[123]: Error in nginx.",
+            "Jan 01 12:00:00 host systemd[1]: Test log entry.",
             "",
         )
 
-        result = await mcp.call_tool(
-            "get_journal_logs", {"unit": "nginx.service", "priority": "err", "since": "today", "lines": 50}
-        )
-
-        assert isinstance(result, tuple)
-        assert len(result) == 2
-        assert isinstance(result[0], list)
-        output = result[0][0].text
-
-        assert "last 50 entries" in output
-        assert "unit=nginx.service" in output
-        assert "priority=err" in output
-        assert "since=today" in output
-
-        # Verify all filters were passed
-        args = mock_execute_command.call_args[0][0]
-        assert "-u" in args
-        assert "nginx.service" in args
-        assert "-p" in args
-        assert "err" in args
-        assert "--since" in args
-        assert "today" in args
-        assert "-n" in args
-        assert "50" in args
-
-    async def test_get_journal_logs_no_entries_found(self, mocker):
-        """Test get_journal_logs when no entries match criteria."""
-        mock_execute_command = mocker.patch("linux_mcp_server.tools.logs.execute_command")
-        mock_execute_command.return_value = (0, "", "")
-
-        result = await mcp.call_tool("get_journal_logs", {})
-
-        assert isinstance(result, tuple)
-        assert len(result) == 2
-        assert isinstance(result[0], list)
-        output = result[0][0].text
-
-        assert "No journal entries found matching the criteria" in output
-
-    async def test_get_journal_logs_command_error(self, mocker):
-        """Test get_journal_logs handles command errors."""
-        mock_execute_command = mocker.patch("linux_mcp_server.tools.logs.execute_command")
-        mock_execute_command.return_value = (1, "", "journalctl: Failed to access journal")
-
-        result = await mcp.call_tool("get_journal_logs", {})
+        result = await mcp.call_tool("get_journal_logs", params)
+        output = await assert_tool_result_structure(result)
```

Comment:
```suggestion
        output =  assert_tool_result_structure(result)
```

we don't need to await on this assert because we have already awaited the result above